src/cli/uuid.test.ts

```diff
@@ -0,0 +1,132 @@
+          labels: { "pepr.dev/uuid": "1234" },
+          annotations: { "pepr.dev/description": "Test annotation" },
+        },
+      },
+    ],
+  });
+
+  const WithLabel = vi.fn().mockReturnValue({ Get });
+  const InNamespace = vi.fn().mockReturnValue({ WithLabel });
+
+  const K8sMock = vi.fn().mockReturnValue({ InNamespace });
+
+  return {
+    ...actual,
+    K8s: K8sMock,
+    kind: actual.kind,
+  };
+});
+
+describe("buildUUIDTable", () => {
+  it("should build table with UUID and description", () => {
+    const deployments: KubernetesListObject<kind.Deployment> = {
+      apiVersion: "v1",
+      kind: "List",
+      items: [
+        {
+          metadata: {
+            name: "test-deploy",
+            labels: { "pepr.dev/uuid": "1234" },
+            annotations: { "pepr.dev/description": "Test annotation" },
+          },
+        },
+      ],
+    };
+
+    const result = buildUUIDTable(deployments);
+    expect(result).toEqual({ "1234": "Test annotation" });
+  });
+
+  it("should include UUID with empty description if none is present", () => {
+    const deployments: KubernetesListObject<kind.Deployment> = {
+      apiVersion: "v1",
+      kind: "List",
+      items: [
+        {
+          metadata: {
+            name: "no-desc",
+            labels: { "pepr.dev/uuid": "5678" },
+          },
+        },
+      ],
+    };
+
+    const result = buildUUIDTable(deployments);
+    expect(result).toEqual({ "5678": "" });
+  });
+
+  it("should skip deployments without UUID", () => {
+    const deployments: KubernetesListObject<kind.Deployment> = {
+      apiVersion: "v1",
+      kind: "List",
+      items: [
+        {
+          metadata: {
+            name: "no-uuid",
+            annotations: { "pepr.dev/description": "Should be ignored" },
+          },
+        },
+      ],
+    };
+
+    const result = buildUUIDTable(deployments);
+    expect(result).toEqual({});
+  });
+});
+describe("getPeprDeploymentsByUUID", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it("should fetch deployments without uuid", async () => {
+    const result = await getPeprDeploymentsByUUID();
+
+    expect(result.items[0].metadata?.labels?.["pepr.dev/uuid"]).toBe("1234");
+    expect(result.items[0].metadata?.annotations?.["pepr.dev/description"]).toBe("Test annotation");
+  });
+
+  it("should fetch deployments with a specific uuid", async () => {
+    const result = await getPeprDeploymentsByUUID("abcd-uuid");
+
+    expect(result.items[0].metadata?.labels?.["pepr.dev/uuid"]).toBe("1234");
+  });
+});
+
+describe("uuid CLI command", () => {
+  let program: Command;
+  let logSpy: ReturnType<typeof vi.spyOn>;
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+    program = new Command();
+    uuid(program);
+    logSpy = vi.spyOn(console, "log").mockImplementation(() => {});
+  });
+
+  it("should log UUID and description when deployment has a UUID", async () => {
+    await program.parseAsync(["node", "test", "uuid"]);
```

Comment:
```suggestion
    await program.parseAsync(["uuid"], { from: "user" });
```

I believe this use of `parseAsync` is more clear, since it only focuses on input we'd expect from a user and does not require `node` or `test` to precede our command.